pkg/recipes/terraform/config/config.go

```diff
@@ -52,6 +60,84 @@ func New(moduleName string, envRecipe *recipes.EnvironmentDefinition, resourceRe
 	}
 }
 
+// getModuleSource is called to get the module source path with credential information
+// if template path represent a private source else returns the provided template path
+func getModuleSource(ctx context.Context, envConfig *recipes.Configuration, templatePath string, armOptions *arm.ClientOptions) (string, error) {
+	var source string
+	var err error
+	if strings.HasPrefix(templatePath, "git::") {
+		source, err = getGitSource(ctx, envConfig, templatePath, armOptions)
+		if err != nil {
+			return "", err
+		}
+	} else {
+		source = templatePath
+	}
+	return source, err
+}
+
+// getGitSource creates a terraform module source in a generic git format with credential information
+// e.g: git::https://<username>:<pat>@<example.com>/<repo_path>
+func getGitSource(ctx context.Context, envConfig *recipes.Configuration, templatePath string, armOptions *arm.ClientOptions) (string, error) {
+	secretStore, err := getSecretStoreID(*envConfig, templatePath)
+	if err != nil {
+		return "", err
+	}
+	if secretStore != "" {
+		secretStoreID, err := resources.ParseResource(secretStore)
+		if err != nil {
+			return "", err
+		}
+		client, err := v20231001preview.NewSecretStoresClient(secretStoreID.RootScope(), &aztoken.AnonymousCredential{}, armOptions)
+		if err != nil {
+			return "", err
+		}
+		secrets, err := client.ListSecrets(ctx, secretStoreID.Name(), map[string]any{}, nil)
+		if err != nil {
+			return "", err
+		}
+		url, err := getGitURL(templatePath)
+		if err != nil {
+			return "", err
+		}
+		var username, pat *string
+		path := "git::https://"
+		user, ok := secrets.Data["username"]
+		if ok {
+			username = user.Value
+			path += fmt.Sprintf("%s:", *username)
+		}
+		token, ok := secrets.Data["pat"]
+		if ok {
+			pat = token.Value
+			path += *pat
+		}
+		path += fmt.Sprintf("@%s", strings.TrimPrefix(url.String(), "https://"))
+		return path, err
+	}
+
+	return templatePath, err
```

Comment:
You could do the other case first: `if secretStore != "" { return templatePath, err }`. Then the rest doesn't need to be within an if case. I find this more readable: **first exit cases then the actual logic**.